scheduler/src/cook/mesos/api.clj

```diff
@@ -2037,6 +2134,30 @@
                    :handler (read-jobs-handler-multiple conn framework-id is-authorized-fn
                                                         retrieve-sandbox-directory-from-agent)}}))
 
+        (c-api/context
+          "/instances/:uuid" [uuid]
+          :path-params [uuid :- s/Uuid]
+          (c-api/resource
+            {:get {:summary "Returns info about a single Job Instance"
+                   :responses {200 {:schema InstanceResponse
+                                    :description "The job instance was returned."}
+                               400 {:description "A non-UUID value was passed."}
+                               403 {:description "The supplied UUID doesn't correspond to a valid job instance."}}
+                   :handler (read-instances-handler-single conn framework-id is-authorized-fn
+                                                           retrieve-sandbox-directory-from-agent)}}))
+
+        (c-api/context
+          "/instances" []
+          (c-api/resource
+            {:get {:summary "Returns info about a set of Job Instances"
```

Comment:
Aside: The word "set" here has some connotations that we may or may not want (unique and unordered). We should clarify our API at some point to address these questions:
1. Is it an error to pass the same UUID multiple times?
2. Are the instance responses returned in the same order as their UUIDs were submitted?